### DISCUSSION TITLE
`json_encoders` is deprecate

### DISCUSSION THREAD

**Discussion by FAbrahamDev** [pydantic/pydantic#10896] [Pydantic V2]
\pydantic\_internal\_generate_schema.py:312: PydanticDeprecatedSince20: `json_encoders` is deprecate
d. See https://docs.pydantic.dev/2.9/concepts/serialization/#custom-serializers for alternatives. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.9/migration/

```python
class BaseResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True, populate_by_name=True, json_schema_extra=json_schema_extra, json_encoders={UUID: lambda v: str(v)},
    )
```

How would i do that: json_encoders={UUID: lambda v: str(v)},
with the new encoders writing style?

**Comment by WilliamStam** [comment]
ours uses 

```
from pydantic import BaseModel as ParentBaseModel
class BaseModel(ParentBaseModel):
    model_config = ConfigDict(
        use_enum_values=True,
        json_encoders={
            datetime: lambda value: value.strftime("%Y-%m-%d %H:%M:%S"),
            date: lambda value: value.strftime("%Y-%m-%d"),
            time: lambda value: value.strftime("%H:%M:%S"),
            Decimal: lambda value: float(value),
        },
    )

```

we try to format all the date fields the same way. Decimals are also returned as strings by default. 

going through and creating serializers for every field is not going to be possible. also many mistakes from oops forgot to update a datetime field format type stuff. 

how are we supposed to do this in the new pydantic ways?

a tried approach is

```
@field_serializer("*", mode="plain", check_fields=False)
    def serialize_types(self, value):
        match value:
            case datetime():
                return value.strftime("%Y-%m-%d %H:%M:%S")
            case date():
                return value.strftime("%Y-%m-%d")
            case time():
                return value.strftime("%H:%M:%S")
            case Decimal():
                return float(value)
            case _:
                return value
```

but im not sure if this works as well

**Comment by Samielakkad** [comment]
For the UUID case, you do not need a custom serializer if the output is JSON. Pydantic v2 already converts `UUID` values in JSON mode:

```python
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class BaseResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True, populate_by_name=True)


class Response(BaseResponse):
    id: UUID


response.model_dump(mode="json")  # UUID becomes a string
response.model_dump_json()
```

If you need non-default formatting, the v2 replacement is a functional serializer. An `Annotated` alias makes the behavior reusable without repeating field names:

```python
from typing import Annotated
from uuid import UUID

from pydantic import PlainSerializer


UUIDText = Annotated[
    UUID,
    PlainSerializer(str, return_type=str, when_used="json"),
]


class Response(BaseResponse):
    id: UUIDText
```

The same pattern can define reusable aliases for `datetime`, `date`, and `Decimal`. A `@field_serializer(*)` on a common base model is also supported and includes subclass fields, but it operates on that model's fields; nested models should use the same base model or annotated aliases. Check `info.mode` in a wildcard serializer if the transformation should only affect JSON output.

Serialization and serializer documentation: https://docs.pydantic.dev/latest/concepts/serialization/#serializers
